### PR TITLE
Fix baseHref, refactor config and build

### PR DESCRIPTION
**Pull request by levithomason**
The base href on the doc site was recently updated.  In local dev, the anchor links are broken as they resolve to absolute urls.  This PR addresses that.

In the way of this was some config/build cleanup and simplification.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,8 +1,3 @@
-# HEADS UP!
-#
-#   'npm run clean' runs rm -rf on all lines in this file.
-#
-
```
Comment (levithomason) [inline .gitignore:5]: cruft

File: config.js [hunk 2]
```diff
@@ -8,15 +8,15 @@ const __DEV__ = env === 'development'
 const __STAGING__ = env === 'staging'
 const __TEST__ = env === 'test'
 const __PROD__ = env === 'production'
-const __BASE__ = '/'
+const __BASE__ = __PROD__ ? '/stardust' : ''
```
Comment (levithomason) [inline config.js:11]: No base href in local dev, which as you'll see in `index.ejs` just omits the base tag altogether.

File: config.js [hunk 3]
```diff
@@ -8,15 +8,15 @@ const __DEV__ = env === 'development'
...
 
 let config = {
   env,
 
   // ----------------------------------
   // Project Structure
   // ----------------------------------
-  path_base: path.resolve(__dirname, '../'),
+  path_base: __dirname,
```
Comment (levithomason) [inline config.js:19]: The config file is a single file in the root now.

File: config.js [hunk 4]
```diff
@@ -52,14 +52,14 @@ config = Object.assign({}, config, {
...
+  compiler_devtool: __DEV__ && 'eval-cheap-module-source-map' || __STAGING__ && 'source-map',
+  compiler_hash_type: __PROD__ ? 'chunkhash' : 'hash',
   compiler_inline_manifest: false,
-  compiler_fail_on_warning: false,
+  compiler_fail_on_warning: (__TEST__ || __PROD__),
   compiler_lint: argv.lint !== false,
   compiler_quiet: false,
   compiler_output_path: paths.base(config.dir_docs_dist),
-  compiler_public_path: __BASE__,
+  compiler_public_path: __PROD__ ? __BASE__ : '/',
```
Comment (levithomason) [inline config.js]: Removed the `/config` directory and 5 files of complexity in favor of 4 lines of conditionals in a single file.  This caught duplicates and unused config as well.

Comment (levithomason) [inline config.js]: Updating to `__BASE__ || '/'`

File: config.js [hunk 5]
```diff
@@ -52,14 +52,14 @@ config = Object.assign({}, config, {
...
+  compiler_devtool: __DEV__ && 'eval-cheap-module-source-map' || __STAGING__ && 'source-map',
+  compiler_hash_type: __PROD__ ? 'chunkhash' : 'hash',
   compiler_inline_manifest: false,
-  compiler_fail_on_warning: false,
+  compiler_fail_on_warning: __TEST__ || __PROD__,
   compiler_lint: argv.lint !== false,
   compiler_quiet: false,
   compiler_output_path: paths.base(config.dir_docs_dist),
-  compiler_public_path: __BASE__,
+  compiler_public_path: __BASE__ || '/',
```
Comment (levithomason) [inline config.js:62]: Removed the /config directory and 5 files of complexity in favor of 4 lines of conditionals in a single file. This caught duplicates and unused config as well.

**Comment by dvdzkwsk** [comment]
👻

**Pull request merged**